API/projectClass/contrastsClass.m

```diff
@@ -105,127 +124,148 @@
-                    end
-                    
-                    if ~isempty(inputBlock.name)
-                        thisContrast.name = inputBlock.name;
-                    end
-                    
-                    if ~isempty(inputBlock.nba)
-                        thisContrast.nba = inputBlock.nba;
-                    end
-                    
-                    if ~isempty(inputBlock.nbs)
-                        thisContrast.nbs = inputBlock.nbs;
-                    end
-                    
-                    if ~isempty(inputBlock.resolution)
-                        thisContrast.resolution = inputBlock.resolution;
-                    end
-                    
-                    if ~isempty(inputBlock.resample)
-                        thisContrast.resample = inputBlock.resample;
-                    end
-                    
-                    if ~isempty(inputBlock.scalefactor)
-                        thisContrast.scalefactor = inputBlock.scalefactor;
-                    end
-                    
-            %end
+            % Raise a warning if we try to set the model as this should be
+            % done elsewhere
+            inputBlock = obj.parseContrastInput(allowedNames,varargin{:});
+            
+            if ~isempty(inputBlock.background)
+                thisContrast.background = inputBlock.background;
+            end
             
-            obj.contrasts{whichContrast} = thisContrast;
+            if ~isempty(inputBlock.data)
+                thisContrast.data = inputBlock.data;
+            end
+            
+            if ~isempty(inputBlock.name)
+                thisContrast.name = inputBlock.name;
+            end
+            
+            if ~isempty(inputBlock.nba)
+                thisContrast.nba = inputBlock.nba;
+            end
+            
+            if ~isempty(inputBlock.nbs)
+                thisContrast.nbs = inputBlock.nbs;
+            end
+            
+            if ~isempty(inputBlock.resolution)
+                thisContrast.resolution = inputBlock.resolution;
+            end
+            
+            if ~isempty(inputBlock.resample)
+                thisContrast.resample = inputBlock.resample;
+            end
+            
+            if ~isempty(inputBlock.scalefactor)
+                thisContrast.scalefactor = inputBlock.scalefactor;
+            end
+            
+            obj.contrasts{contrastIndex} = thisContrast;
             
         end
         
         function contrastNames = getAllContrastNames(obj)
+            % Get the names of all contrasts defined in the class.
+            %
+            % contrasts.getAllContrastNames()
             nContrasts = obj.numberOfContrasts;
             contrastNames = cell(1,nContrasts);
             
             if nContrasts == 0
-                error('Can''t get contrast names when no contrasts defined')
+                throw(invalidValue('Can''t get contrast names when no contrasts defined'));
```

Comment:
Do you know why this throws an error instead of just returning an empty array?

Reply:
Good point, there doesn't appear to be any reason to me, as this routine is only used within the class and we check if the result is empty when we call it. I'll remove this clause.